src/benchmarks/micro/libraries/System.Numerics.Tensors/Perf_SignedIntegerTensorPrimitives.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+using System.Linq;
+
+namespace System.Numerics.Tensors.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD, Categories.JIT)]
+    [GenericTypeArguments(typeof(sbyte))]
+    [GenericTypeArguments(typeof(short))]
+    [GenericTypeArguments(typeof(int))]
+    [GenericTypeArguments(typeof(long))]
+    public class Perf_SignedIntegerTensorPrimitives<T>
+        where T : unmanaged, IBinaryInteger<T>, IMinMaxValue<T>
+    {
+        [Params(3, 64 + 7, 128 + 7, 256 + 7, 512 + 7, 3 * 512, 6 * 512)]
+        public int BufferLength;
+
+        private const int ShiftAmount = 3;
+
+        private T[] _source1;
+        private T[] _source2;
+        private T _scalar1;
+        private T[] _ones;
+        private T[] _destination;
+
+        [GlobalSetup]
+        public void Init()
+        {
+            _source1 = ValuesGenerator.Array<T>(BufferLength, seed: 42);
+            _source2 = ValuesGenerator.Array<T>(BufferLength, seed: 43);
+            _scalar1 = ValuesGenerator.Value<T>(seed: 44);
+            _ones = Enumerable.Repeat(T.One, BufferLength).ToArray();
+            _destination = new T[BufferLength];
+        }
+
+        #region Unary Operations
+        [Benchmark]
+        public void Abs() => TensorPrimitives.Abs<T>(_source1, _destination);
```

Comment:
One other question I have. What is the runtime per operation? If these are anything like the Vector tests, we would likely want to increase the number of calls per operation to try and ensure these tests are not in the single nanosecond range per op. This will help with noise in the benchmark.

Reply:
They appear to be sitting in the double digit range:

| Method              | BufferLength | Mean          | Error         | StdDev        | Median        | Min           | Max          | Allocated |
|-------------------- |------------- |--------------:|--------------:|--------------:|--------------:|--------------:|-------------:|----------:|
| **Abs**                 | **8**            |     **19.347 ns** |     **3.2394 ns** |     **3.6006 ns** |     **17.895 ns** |     **15.898 ns** |     **29.04 ns** |         **-** |
| LeadingZeroCount    | 8            |     13.417 ns |     1.1495 ns |     1.1804 ns |     12.829 ns |     12.415 ns |     16.91 ns |         - |
| OnesComplement      | 8            |     18.505 ns |     2.0972 ns |     2.3310 ns |     17.982 ns |     15.733 ns |     24.80 ns |         - |
| PopCount            | 8            |     14.791 ns |     0.5889 ns |     0.6782 ns |     14.669 ns |     13.744 ns |     16.16 ns |         - |
| ShiftLeft           | 8            |     13.011 ns |     0.3786 ns |     0.3888 ns |     13.126 ns |     12.361 ns |     13.65 ns |         - |
| TrailingZeroCount   | 8            |     14.704 ns |     3.1315 ns |     3.4807 ns |     13.711 ns |     11.160 ns |     21.79 ns |         - |
| BitwiseAnd_Vector   | 8            |     15.643 ns |     1.0767 ns |     1.1057 ns |     15.167 ns |     14.795 ns |     19.06 ns |         - |
| BitwiseAnd_Scalar   | 8            |     17.004 ns |     6.6236 ns |     7.6278 ns |     11.559 ns |     10.845 ns |     29.39 ns |         - |
| Divide_Vector       | 8            |    247.337 ns |    13.7346 ns |    15.8168 ns |    250.798 ns |    217.208 ns |    264.92 ns |         - |
| Divide_Scalar       | 8            |    226.290 ns |    13.6104 ns |    15.6737 ns |    228.483 ns |    187.753 ns |    250.10 ns |         - |
| Max_Vector          | 8            |     15.919 ns |     0.8274 ns |     0.8496 ns |     15.642 ns |     14.856 ns |     18.16 ns |         - |
| Max_Scalar          | 8            |      9.886 ns |     0.2134 ns |     0.2096 ns |      9.896 ns |      9.586 ns |     10.48 ns |         - |
| MaxMagnitude_Vector | 8            |     20.081 ns |     0.8079 ns |     0.8980 ns |     20.055 ns |     19.040 ns |     22.51 ns |         - |
| MaxMagnitude_Scalar | 8            |     33.059 ns |     7.0799 ns |     8.1532 ns |     34.346 ns |     19.363 ns |     43.28 ns |         - |
| SumOfMagnitudes     | 8            |     18.353 ns |     4.4960 ns |     5.1776 ns |     18.917 ns |     11.665 ns |     25.73 ns |         - |
| **Abs**                 | **3079**         |  **1,184.183 ns** |   **132.7364 ns** |   **152.8595 ns** |  **1,230.750 ns** |    **939.998 ns** |  **1,404.89 ns** |         **-** |
| LeadingZeroCount    | 3079         |  2,623.800 ns |   259.6078 ns |   298.9648 ns |  2,523.757 ns |  2,224.539 ns |  3,133.47 ns |         - |
| OnesComplement      | 3079         |    510.886 ns |    61.9353 ns |    71.3248 ns |    480.912 ns |    435.789 ns |    655.61 ns |         - |
| PopCount            | 3079         |  1,720.535 ns |   198.1776 ns |   228.2216 ns |  1,712.257 ns |  1,399.724 ns |  2,077.84 ns |         - |
| ShiftLeft           | 3079         |    420.811 ns |    26.0210 ns |    28.9223 ns |    411.772 ns |    386.873 ns |    489.19 ns |         - |
| TrailingZeroCount   | 3079         |  1,607.124 ns |    59.6521 ns |    68.6955 ns |  1,586.087 ns |  1,511.768 ns |  1,756.80 ns |         - |
| BitwiseAnd_Vector   | 3079         |    705.526 ns |    58.8464 ns |    65.4076 ns |    695.991 ns |    614.916 ns |    861.25 ns |         - |
| BitwiseAnd_Scalar   | 3079         |    684.698 ns |   132.5878 ns |   152.6883 ns |    617.481 ns |    473.334 ns |    994.09 ns |         - |
| Divide_Vector       | 3079         | 18,454.394 ns | 1,587.0094 ns | 1,827.6024 ns | 18,578.829 ns | 15,040.439 ns | 21,270.93 ns |         - |
| Divide_Scalar       | 3079         | 14,145.835 ns |   413.4350 ns |   424.5673 ns | 14,078.132 ns | 13,556.851 ns | 14,988.75 ns |         - |
| Max_Vector          | 3079         |    518.488 ns |    13.6749 ns |    14.6320 ns |    515.133 ns |    499.432 ns |    544.60 ns |         - |
| Max_Scalar          | 3079         |    551.734 ns |    84.4518 ns |    93.8679 ns |    547.237 ns |    381.233 ns |    713.14 ns |         - |
| MaxMagnitude_Vector | 3079         |  1,590.950 ns |   389.9779 ns |   449.0991 ns |  1,410.396 ns |  1,123.538 ns |  2,651.76 ns |         - |
| MaxMagnitude_Scalar | 3079         |  1,050.634 ns |   145.3134 ns |   167.3431 ns |  1,028.143 ns |    845.030 ns |  1,410.43 ns |         - |
| SumOfMagnitudes     | 3079         |    441.706 ns |     4.3370 ns |     3.8446 ns |    442.291 ns |    436.049 ns |    446.63 ns |         - |